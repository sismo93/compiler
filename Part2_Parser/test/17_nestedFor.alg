begin
  a:= 32;
  b:=2;
  for j from 458-55 by b to 100+232-2 do
    print(a);
    for i from j by 5 to 10 do
     print(b)
    endwhile
  endwhile

end
