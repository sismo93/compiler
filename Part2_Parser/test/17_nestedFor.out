Rules used : 1 3 4 7 13 14 15 18 23 21 5 4 7 13 14 15 18 23 21 5 4 10 46 14 15 18 23 20 25 15 18 23 21 14 15 17 23 21 14 15 18 23 20 24 15 18 23 20 25 15 18 23 21 3 4 11 47 5 4 10 46 14 15 17 23 21 14 15 18 23 21 14 15 18 23 21 3 4 11 47 6 6 6 
 Here are all the grammar used with more detail : 

<Program> -> begin <Code> end

<Code> -> <InstList>

<InstList> -> <Instruction> <InstListPrim>

<Instruction> -> <Assign>

<Assign> -> [VarName] := <ExprArith>

<ExprArith> -> <Term> <ExprArith_prim>

<Term> -> <Atom> <prodPrim>

<Atom> -> NUMBER

<prodPrim> -> EPSILON

<ExprArith_prim> -> EPSILON

<InstListPrim> -> SEMICOLON <instList>

<InstList> -> <Instruction> <InstListPrim>

<Instruction> -> <Assign>

<Assign> -> [VarName] := <ExprArith>

<ExprArith> -> <Term> <ExprArith_prim>

<Term> -> <Atom> <prodPrim>

<Atom> -> NUMBER

<prodPrim> -> EPSILON

<ExprArith_prim> -> EPSILON

<InstListPrim> -> SEMICOLON <instList>

<InstList> -> <Instruction> <InstListPrim>

<Instruction> -> <For>

<For> -> for [VarName] from <exprArith> by <exprArith> to <exprArith> do <code> endwhile

<ExprArith> -> <Term> <ExprArith_prim>

<Term> -> <Atom> <prodPrim>

<Atom> -> NUMBER

<prodPrim> -> EPSILON

<ExprArith_prim> -> <AddOp> <Term> <ExprArith_prim>

<addOp> -> -

<Term> -> <Atom> <prodPrim>

<Atom> -> NUMBER

<prodPrim> -> EPSILON

<ExprArith_prim> -> EPSILON

<ExprArith> -> <Term> <ExprArith_prim>

<Term> -> <Atom> <prodPrim>

<Atom> -> VARNAME

<prodPrim> -> EPSILON

<ExprArith_prim> -> EPSILON

<ExprArith> -> <Term> <ExprArith_prim>

<Term> -> <Atom> <prodPrim>

<Atom> -> NUMBER

<prodPrim> -> EPSILON

<ExprArith_prim> -> <AddOp> <Term> <ExprArith_prim>

<addOp> -> +

<Term> -> <Atom> <prodPrim>

<Atom> -> NUMBER

<prodPrim> -> EPSILON

<ExprArith_prim> -> <AddOp> <Term> <ExprArith_prim>

<addOp> -> -

<Term> -> <Atom> <prodPrim>

<Atom> -> NUMBER

<prodPrim> -> EPSILON

<ExprArith_prim> -> EPSILON

<Code> -> <InstList>

<InstList> -> <Instruction> <InstListPrim>

<Instruction> -> <Print>

<Print> -> print ( <VarName> )

<InstListPrim> -> SEMICOLON <instList>

<InstList> -> <Instruction> <InstListPrim>

<Instruction> -> <For>

<For> -> for [VarName] from <exprArith> by <exprArith> to <exprArith> do <code> endwhile

<ExprArith> -> <Term> <ExprArith_prim>

<Term> -> <Atom> <prodPrim>

<Atom> -> VARNAME

<prodPrim> -> EPSILON

<ExprArith_prim> -> EPSILON

<ExprArith> -> <Term> <ExprArith_prim>

<Term> -> <Atom> <prodPrim>

<Atom> -> NUMBER

<prodPrim> -> EPSILON

<ExprArith_prim> -> EPSILON

<ExprArith> -> <Term> <ExprArith_prim>

<Term> -> <Atom> <prodPrim>

<Atom> -> NUMBER

<prodPrim> -> EPSILON

<ExprArith_prim> -> EPSILON

<Code> -> <InstList>

<InstList> -> <Instruction> <InstListPrim>

<Instruction> -> <Print>

<Print> -> print ( <VarName> )

<InstListPrim> -> EPSILON

<InstListPrim> -> EPSILON

<InstListPrim> -> EPSILON

